### DISCUSSION TITLE
MastoHttpError: Error processing thumbnail for uploaded media

### DISCUSSION THREAD

**Discussion by individual8** [neet/masto.js#977] [Q&A] [answered]
I stumbled into another error that is unresolved. It seems like Mastodon cannot swallow my image upload.

Here's my code:
```
try {
    // upload media
    const photo = await fetch(mediaUrl + '=w1500-h1500');
    const thumbnail = await fetch(mediaUrl + '=w750-h750');
    const attachment = await masto.v2.media.create({
      file: photo.blob(),
      thumbnail: thumbnail.blob()
    });
    try {
      // post status
      const status = await masto.v1.statuses.create({
        status: caption,
        visibility: "public",
        mediaIds: [attachment.id],
      });
      // save posted photo to database
      await updatePhotosPosted(service, photosPosted, photoToBePosted);
    } catch (e) {
      functions.logger.log(`Error publishing photo on Mastodon.`, e);
    }
  } catch (e) {
    functions.logger.log(`Error uploading photo to Mastodon.`, e);
  }
```

I tried first w/o specifying an thumbnail. And then with an thumbnail not bigger than 750x750 px. Both end up in the same 500 (internal server error).

**Comment by individual8** [comment]
@neet Is that something than can be fixed?

**Reply by neet** [reply]
@individual8  Unfortunately, I can't identify the cause from your report. I assume `thumbnail` is supposed to be used with a video so cannot be combined with an image, but since you have also tried without a thumbnail and ended up with the same result, I believe the problem is from a different factor.

Can you try the same thing in [different instances](https://joinmastodon.org/servers)? Since each instance has its own limit for media size to be uploaded, I can't say 750x750 is small enough.

**Reply by neet** [reply, marked as answer]
I didn't notice initially, but apparently, you seem to have forgotten `await`-ing the conversion from a `File` to a `Blob`. Try changing as the snippet below:

```diff
const attachment = await masto.v2.media.create({
-  file: photo.blob(),
-  thumbnail: thumbnail.blob()
+  file: await photo.blob(),
+  thumbnail: await thumbnail.blob()
});
```
